### ISSUE TITLE
List is incompatible with Sequence in parameter

### ISSUE THREAD

**Issue by isac322** [python/mypy#7992]
* __Are you reporting a bug, or opening a feature request?__

A Bug maybe?

* __Please insert below the code you are checking with mypy__

`test.py`:
```python
from typing import Sequence, List

class A:
    def test(self, t: Sequence[int]) -> Sequence[str]:
        pass

class B(A):
    def test(self, t: List[int]) -> List[str]:
        pas
```

* __What is the actual output?__

`mypy test.py`
```
test.py:8: error: Argument 1 of "test" is incompatible with supertype "A"; supertype defines the argument type as "Iterable[int]"
Found 1 error in 1 file (checked 1 source file)
```

* __What is the output you expect?__

`mypy test.py`
```
Success: no issues found in 1 source file
```

* __What are the versions of mypy and Python you are using?__

  * mypy: 0.740
  * python: 2.7.13 and 3.7.6 (tested both)

* __Do you see the same issue after installing mypy from Git master?__

Yes (on 989626a8)

**Comment by JukkaL** [comment]
This signature violates Liskov substitution principle (https://en.wikipedia.org/wiki/Liskov_substitution_principle). You can use `# type: ignore[override]` to silence the error (add it to the line that genreates the error) if you don't care about the unsafety.
